test/built-ins/GeneratorPrototype/next/context-constructor-invocation.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 25.2
+description: >
+    When a generator function is invoked as a constructor, its context is the
+    newly-created iterable object.
+---*/
+
+var context;
+function* g() { context = this; }
```

Comment:
Access to `this` in new'ed generators throws a ReferenceError. (Generators are created as derived constructor functions.)